### PR TITLE
Improved UI flow + added a settings template

### PR DESCRIPTION
**Pull request by ShafSpecs**
# Description

This PR improves general flow of Remix Dev Tools including utilisation of JSDocs for props, updating relevant packages and overall aesthetics. 

The second goal of this PR is adding a settings tab. A non-breaking concept to help improve DX and co-locate multiple configurations in one place without ever having to edit the source code.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Most changes are aesthetics (UI improvements) and can't be directly tested. The settings concept has been tested thoroughly (it is a blank window that fits in seamlessly with the others).

# Checklist:

- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] New and existing unit tests pass locally with my changes

# Todos 

Before this is ready for review and possible merging, here are the tasks left:

- ~~Provide a horizontal scroll behaviour for the tab bar at the top (When the screen is shrunk, the tabs should be scrollable like VSCode instead of text wrapping)~~
- ~~Improve spacings~~

### CODE REVIEW

File: src/RemixDevTools/RemixDevTools.tsx [hunk 1]
```diff
@@ -42,22 +46,23 @@ const RemixDevTools = ({ defaultOpen, position }: Props) => {
...
-            "rdt-hover:cursor-pointer rdt-hover:ring-2 rdt-ring-slate-600"
+            "rdt-hover:cursor-pointer rdt-hover:ring-2 rdt-ring-slate-600",
           )}
         />
       </div>
       <div
         style={{ zIndex: 9998, height }}
         className={clsx(
-          "rdt-duration-600 rdt-fixed rdt-bottom-0 rdt-left-0 rdt-box-border rdt-flex rdt-w-screen rdt-resize-y rdt-flex-col rdt-overflow-auto rdt-bg-[#212121] rdt-text-white rdt-opacity-0 rdt-transition-all",
+          "remix-dev-tools-window rdt-duration-600 rdt-fixed rdt-bottom-0 rdt-left-0 rdt-box-border rdt-flex rdt-w-screen rdt-resize-y rdt-flex-col rdt-overflow-auto rdt-bg-[#212121] rdt-text-white rdt-opacity-0 rdt-transition-all",
```
Comment (AlemTuzlak) [inline src/RemixDevTools/RemixDevTools.tsx:65]: You can remove this class and the css in the css file and just remove the rdt-resize-y class from here, I left it in by accident, that should remove the resize on the bottom right

**Review by AlemTuzlak — COMMENTED** [review]
Can you merge the latest dev branch into yours, I changed the target from main to dev so I can do some changes before I merge it to main

File: src/RemixDevTools/RemixDevTools.tsx (hunk 1 above)
Comment (ShafSpecs) [reply src/RemixDevTools/RemixDevTools.tsx:73]: 👍

**Comment by ShafSpecs** [comment]
There's a bug with the new changes introduced in `dev` branch. Lemme fix it first before merging.

The issue is that if `defaultOpen` is set to true in the Remix app, the dev window can never close, but if it's false, it can open and close

**Comment by ShafSpecs** [comment]
The problematic code:

```ts
const isOpen = useMemo(
    () => defaultOpen || persistOpen,
    [persistOpen, defaultOpen]
  );
```

It would always return true, i.e `isOpen` can never evaluate to false

**Comment by ShafSpecs** [comment]
@AlemTuzlak is the 24-hide-devtools branch going to be merged into dev upstream?

**Comment by AlemTuzlak** [comment]
@ShafSpecs yes, I had to refactor a bit, also fixed the issue that you wrote about

**Comment by ShafSpecs** [comment]
> @ShafSpecs yes, I had to refactor a bit, also fixed the issue that you wrote about

Alright, would wait for the merge then push my changes 👍

**Comment by AlemTuzlak** [comment]
@ShafSpecs will let you know, after reviewing your code i realized i meed to refactor, will add more stuff on top so best to wait a bit, will probably be done by tomorrow. Want to make it easier to extend

**Comment by ShafSpecs** [comment]
@AlemTuzlak updated 👍😄

**Comment by AlemTuzlak** [comment]
@ShafSpecs Would you mind closing this PR and opening a new one based off of the latest main, I can't track of what was changed by me and what was done by you on this one unfortunately

**Pull request closed without merging**